File: src/main/java/br/com/leonardoferreira/jirareport/domain/vo/EstimateIssue.java
package br.com.leonardoferreira.jirareport.domain.vo;

import br.com.leonardoferreira.jirareport.domain.embedded.Changelog;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@ToString(of = "key")
public class EstimateIssue {
    private String key;

    private String issueType;

    private String creator;

    private String system;

    private String epic;

    private String summary;

    private String estimated;

    private String project;

    private LocalDateTime startDate;

    private LocalDate estimateDateAvg;

    private LocalDate estimateDatePercentile50;

    private LocalDate estimateDatePercentile75;

    private LocalDate estimateDatePercentile90;

    private Long leadTime;

    private LocalDateTime created;

    private String priority;

    private List<Changelog> changelog;

    private Long impedimentTime;
}

File: src/main/kotlin/br/com/jiratorio/domain/estimate/EstimateIssue.kt
package br.com.jiratorio.domain.estimate

import br.com.jiratorio.domain.entity.embedded.Changelog
import java.time.LocalDate
import java.time.LocalDateTime

data class EstimateIssue(

    val key: String,

    val summary: String,

    val startDate: LocalDateTime,

    var estimateDateAvg: LocalDate = startDate.toLocalDate(),

    var estimateDatePercentile50: LocalDate = startDate.toLocalDate(),

    var estimateDatePercentile75: LocalDate = startDate.toLocalDate(),

    var estimateDatePercentile90: LocalDate = startDate.toLocalDate(),

    val leadTime: Long,

    val created: LocalDateTime,

    val issueType: String? = null,

    val creator: String? = null,

    val estimate: String? = null,

    val system: String? = null,

    val project: String? = null,

    val epic: String? = null,

    val priority: String? = null,

    val changelog: List<Changelog>,

    val impedimentTime: Long

)
